org.eclipse.xtext/src/org/eclipse/xtext/diagnostics/AbstractDiagnostic.java

```diff
@@ -58,6 +58,25 @@ public int getLine() {
 			return node.getStartLine();
 		return -1;
```

Comment:
mind the contract
> '0' indicates 'no line information'

Reply:
Maybe we should reverse the contract: valid line informations is 1 or higher.
-1, 0, null all indicate absent line information :/